src/services/google-analytics/index.ts

```diff
@@ -16,46 +16,75 @@
 
 import loadGoogleAnalytics from './loadGoogleAnalytics';
 
+type TrackPageView = {
+  page_title?: string;
+  page_location?: string;
+  page_path: string;
+};
+
+type TrackEvent = {
+  event_action: string;
+  event_category: string;
+  event_label?: string;
+  event_value?: number;
+};
+
+type SetDimension = {
+  event_name: string;
+  dimension_key: string;
+  dimension_value: string | null;
+};
+
 /**
  * The interface of the GoogleAnalytics class below
  * replicates the interface of the ReactGA library that we were using before
  */
 
 class GoogleAnalytics {
+  static googleAnalyticsKey: string;
+
   static initialize(googleAnalyticsKey: string) {
     loadGoogleAnalytics(googleAnalyticsKey);
+    this.googleAnalyticsKey = googleAnalyticsKey;
   }
 
-  static ga(...args: any[]) {
-    window.ga(...args);
-  }
-
-  // referencee: example from Google's docs for analytics.js
-  // https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#hitType
-  static pageview(pagePath: string) {
-    window.ga('send', {
-      hitType: 'pageview',
-      page: pagePath
+  // reference: example from Google's docs for gtag.js
+  // https://developers.google.com/analytics/devguides/collection/gtagjs/pages
+  static pageview(params: TrackPageView) {
+    window.gtag('event', 'page_view', {
+      send_to: this.googleAnalyticsKey,
+      ...params
     });
   }
```

Comment:
There are slightly different instructions for single-page applications [here](https://developers.google.com/analytics/devguides/collection/gtagjs/single-page-applications#measure_virtual_pageviews).

Also, is the `send_to` option really necessary? We are always using the same ga measurement id

Reply:
`...params` no longer makes any sense; could as well be just `params`

Reply:
That's done